components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: var(--d2l-button-spacing) !important;
+				margin-bottom: var(--d2l-button-spacing) !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: var(--d2l-button-spacing) !important;
+				margin-right: 0 !important;
+			}
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this._reposition);
+		window.removeEventListener('scroll', this._reposition);
+	}
+
+	render() {
+		return html`
+			<lit-media-query
+  				.query="(max-height: ${this.minHeight})"
+  				@changed="${this._handleMediaQuery}">
+			</lit-media-query>
+			<div class="d2l-floating-buttons-container">
+				<div class="d2l-floating-buttons-inner-container">
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"></div>
+		`;
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
+		this.updateComplete.then(() => {
+
+			this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+			this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+
+			window.addEventListener('resize', this._reposition);
+			window.addEventListener('scroll', this._reposition);
+
+			this._isRTL = (getComputedStyle(this._container).direction === 'rtl');
+			this.updateComplete.then(() => {
+				this._reposition();
+				let prevDocumentHeight = document.body.offsetHeight;
+				setInterval(() => {
```

Comment:
It would be good to determine the purpose of this code (I know this isn't new) - it's a significant performance concern because it's forcing measurement (due to `document.body.offsetHeight`) every 100ms indefinitely. It could cause forced layout/reflow if this runs in a frame with stale layout. Once we get this component working, I'm really hoping that we can make some good improvements to how this works.

Reply:
... this code may be trying to detect DOM mutation - such as addition or removal of content above the buttons that may cause the buttons to float or stop floating. Once we can explore `position: sticky;` and/or resize-observer,  this whole problem may be solved for us.

Reply:
Resize Observers don't seem to be widely implemented yet (see. https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). However, I think the resize event may be covering that. This may be for the "add" and "remove" content detection. we may be able to get away with a Mutation Observer.  (see. https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)

I could try it out unless we want to avoid that for now.

Reply:
I think it would be good to port the existing logic, making the necessary adjustments to get it working with the Lit way of doing things, and getting a nice visual-diff baseline for it. The assumption being that it isn't a big lift to get the existing logic running. If that turns out not to be the case, then yeah, let's talk about sticky positioning and resize-observer sooner rather than later.  (note: there is a resize-observer polyfil that uses native when possible, and falls back to standard resize + mutation observers + transition events)

Reply:
I've hit on most of the other comments in the latest commit, this one I'll need you to walk me through a bit. it looks like a few options, and we'll need to do a bit of research on what one is best suited to us.  It seems you already have a good idea in-mind, and I'd like to know all about it.